uaclient/entitlements/base.py

```diff
@@ -32,12 +32,15 @@ class UAEntitlement(metaclass=abc.ABCMeta):
     # Optional URL for top-level product service information
     help_doc_url = None  # type: str
 
-    #  Whether to assume yes to any messaging prompts
+    # Whether to assume yes to any messaging prompts
     assume_yes = False
 
     # Wheter that entitlement is in beta stage
     is_beta = False
 
+    # List of services that if enabled, will block enabling the service
+    blocking_entitlements = []  # type: List[str]
```

Comment:
We probably minimally want to make this a class method with the @property decorator. The reason being we would like the ability to override this option using feature overrides in configuration.  This primitive probably makes sense as a separate method. But I would like to look at integrating this method/property call into the static_affordances checks for a service. Instead of a separate call from can_enable here's a stale pr I started putting together a while back to solve this issue. But didnt have time to finish it.  Feel free to use any of it if you think it is applicable. https://github.com/canonical/ubuntu-advantage-client/pull/1114